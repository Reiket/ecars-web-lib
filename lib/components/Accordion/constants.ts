export const ACCORDION_TEST_ID = 'accordionTestId';
