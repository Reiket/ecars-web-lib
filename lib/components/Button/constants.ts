import type {ButtonHTMLAttributes} from 'react';

export const BUTTON_SIZE = Object.freeze({
  BIG: 'big',
  SMALL: 'small',
});

export const BUTTON_COLOR = Object.freeze({
  GREEN: 'green',
  WHITE: 'white',
  GRAY: 'gray',
});


export type ButtonAttributesType = ButtonHTMLAttributes<HTMLButtonElement>['type']
export type ButtonSizeType = (typeof BUTTON_SIZE)[keyof typeof BUTTON_SIZE];
export type ButtonColorType = (typeof BUTTON_COLOR)[keyof typeof BUTTON_COLOR];

export const BUTTON_TEST_ID = 'buttonTestId';