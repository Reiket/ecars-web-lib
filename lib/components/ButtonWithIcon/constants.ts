export const LEFT_ICON_BUTTON_TEST_ID = 'leftIconButtonTestId';
export const RIGHT_ICON_BUTTON_TEST_ID = 'rightIconButtonTestId';
