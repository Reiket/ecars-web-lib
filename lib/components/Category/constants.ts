export const CATEGORIES_LIST = Object.freeze({
  GUIDES: 'guides',
  REVIEWS: 'reviews',
  NEWS: 'news',
});

export type CategoriesListType = (typeof CATEGORIES_LIST)[keyof typeof CATEGORIES_LIST];

export const CATEGORY_TEST_ID = 'categoryTestId';
