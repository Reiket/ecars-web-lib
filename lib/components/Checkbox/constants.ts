import type {ElementProps} from '@/services/types';

export interface CheckboxProps extends ElementProps {
  name?: string;
  id?: string;
  disabled?: boolean;
  onChange?: (checked: boolean) => void;
  checked?: boolean;
  hasError?: boolean;
}

export const CHECKBOX_TEST_ID = 'checkboxTestId';
export const CHECKBOX_TEST_INPUT_ID = 'checkboxTestInputId';
