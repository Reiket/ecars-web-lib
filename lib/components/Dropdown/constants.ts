import type {ReactNode} from 'react';

export interface DropdownOption {
  value: string;
  label?: ReactNode;
}

export interface DropdownProps {
  onSelect: (value: string) => void;
  category: string;
  options: DropdownOption[];
}

export const DROPDOWN_TEST_ID = 'dropdownTestId';

export const DROPDOWN_THEME = Object.freeze({
  DARK: 'dark',
  LIGHT: 'light',
});

export type DropdownThemeType = (typeof DROPDOWN_THEME)[keyof typeof DROPDOWN_THEME];
