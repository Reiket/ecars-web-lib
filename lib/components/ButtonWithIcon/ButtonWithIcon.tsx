import type {ComponentType, FC, ReactNode} from 'react';
import type {ButtonProps} from '@/components/Button/Button';
import {Button} from '@/components/Button/Button';
import {LEFT_ICON_BUTTON_TEST_ID, RIGHT_ICON_BUTTON_TEST_ID} from '@/components/ButtonWithIcon/constants';

interface IconProps {
  className?: string;
}

export interface ButtonIconProps<T> extends ButtonProps {
  children: ReactNode;
  RightIcon?: ComponentType<T>;
  LeftIcon?: ComponentType<T>;
  className?: string;
}

export const ButtonWithIcon: FC<ButtonIconProps<IconProps>> = ({
  RightIcon,
  LeftIcon,
  children,
  className,
  ...props
}) => (
  <Button
    withIcon
    {...props}
  >
    {!!LeftIcon && (
      <LeftIcon
        className={className}
        data-testid={LEFT_ICON_BUTTON_TEST_ID}
      />
    )}
    {children}
    {!!RightIcon && (
      <RightIcon
        className={className}
        data-testid={RIGHT_ICON_BUTTON_TEST_ID}
      />
    )}
  </Button>
);
