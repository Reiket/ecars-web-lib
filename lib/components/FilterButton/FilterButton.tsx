import type {FC} from 'react';
import {ButtonWithIcon} from '@/components/ButtonWithIcon/ButtonWithIcon';
import {Icons} from '@/services/icons';
import type {ElementProps} from '@/services/types';

interface Props extends ElementProps {
  children: string;
  onClick?: () => void;
}

export const FilterButton: FC<Props> = ({children, onClick, block}) => (
  <ButtonWithIcon
    type="button"
    block={block}
    onClick={onClick}
    className="button__close"
    RightIcon={Icons.Close}
    size="small"
    color="gray"
  >
    {children}
  </ButtonWithIcon>
);
