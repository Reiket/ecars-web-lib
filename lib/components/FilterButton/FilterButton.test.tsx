import {fireEvent, render, screen} from '@testing-library/react';
import {FilterButton} from './FilterButton';
import {RIGHT_ICON_BUTTON_TEST_ID} from '@/components/ButtonWithIcon/constants';

const defaultProps = {
  onClick: vi.fn(),
};

describe('FilterButton component', () => {
  test('renders the filter component correctly', () => {
    const {container} = render(<FilterButton {...defaultProps}>Close</FilterButton>);
    expect(screen.getByText('Close')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  test('renders RightIcon component', () => {
    render(<FilterButton {...defaultProps}>Close</FilterButton>);
    expect(screen.getByTestId(RIGHT_ICON_BUTTON_TEST_ID)).toBeInTheDocument();
  });

  test('calls onClick when clicked', () => {
    render(<FilterButton {...defaultProps}>Close</FilterButton>);
    fireEvent.click(screen.getByRole('button'));
    expect(defaultProps.onClick).toHaveBeenCalledTimes(1);
  });
});
