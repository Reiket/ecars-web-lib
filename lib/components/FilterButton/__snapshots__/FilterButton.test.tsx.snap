// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FilterButton component > renders the filter component correctly 1`] = `
<div>
  <button
    class="button button--small button--gray button--icon"
    data-testid="buttonTestId"
    type="button"
  >
    Close
    <svg
      class="button__close"
      data-testid="rightIconButtonTestId"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M278.6 256l68.2-68.2c6.2-6.2 6.2-16.4 0-22.6-6.2-6.2-16.4-6.2-22.6 0L256 233.4l-68.2-68.2c-6.2-6.2-16.4-6.2-22.6 0-3.1 3.1-4.7 7.2-4.7 11.3 0 4.1 1.6 8.2 4.7 11.3l68.2 68.2-68.2 68.2c-3.1 3.1-4.7 7.2-4.7 11.3 0 4.1 1.6 8.2 4.7 11.3 6.2 6.2 16.4 6.2 22.6 0l68.2-68.2 68.2 68.2c6.2 6.2 16.4 6.2 22.6 0 6.2-6.2 6.2-16.4 0-22.6L278.6 256z"
      />
    </svg>
  </button>
</div>
`;

exports[`FilterButton component renders the filter component correctly 1`] = `
<div>
  <button
    class="button button--small button--gray button--icon"
    data-testid="buttonTestId"
    type="button"
  >
    Close
    <svg
      class="button__close"
      data-testid="rightIconButtonTestId"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M278.6 256l68.2-68.2c6.2-6.2 6.2-16.4 0-22.6-6.2-6.2-16.4-6.2-22.6 0L256 233.4l-68.2-68.2c-6.2-6.2-16.4-6.2-22.6 0-3.1 3.1-4.7 7.2-4.7 11.3 0 4.1 1.6 8.2 4.7 11.3l68.2 68.2-68.2 68.2c-3.1 3.1-4.7 7.2-4.7 11.3 0 4.1 1.6 8.2 4.7 11.3 6.2 6.2 16.4 6.2 22.6 0l68.2-68.2 68.2 68.2c6.2 6.2 16.4 6.2 22.6 0 6.2-6.2 6.2-16.4 0-22.6L278.6 256z"
      />
    </svg>
  </button>
</div>
`;
